index.js

```diff
@@ -1,14 +1,61 @@
 #! /usr/bin/env node
 'use strict';
 
-/*
+const path = require('path');
+const lodash = require('lodash');
 const utils = require('./lib/utils');
-const DDFRoot = require('./lib/data/root');
-const ddfRoot = new DDFRoot(utils.ddfRootFolder);
+const fileUtils = require('./lib/utils/file');
+const DdfIndexGenerator = require('./lib/ddf-definitions/ddf-index-generator');
+const GenericData = require('./lib/ddf-definitions/generic-data');
+const ddfRules = require('./lib/ddf-rules');
+const logger = utils.logger;
 
-ddfRoot.check(() => {
-  ddfRoot.getDdfDirectoriesDescriptors().forEach(dd => {
-    console.log(`${dd.dir} ${dd.isEmpty} ${dd.isDDF}`);
+if (utils.settings.isIndexGenerateMode === true) {
+  const ddfIndexGenerator = new DdfIndexGenerator(utils.ddfRootFolder);
+
+  /*eslint no-console: [2, { allow: ["log"] }] */
+  ddfIndexGenerator.getCsv((err, csvContent) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+
+    const file = path.resolve(utils.ddfRootFolder, 'ddf--index.csv');
+
+    fileUtils.writeFile(file, csvContent, _err => {
+      if (_err) {
+        console.log(err);
```

Comment:
It seems that you wanted to log another error 
`_err` instead of `err`

Try to give them meaningful names just to avoid such typos